pkg/notifiers/format.go

```diff
@@ -204,6 +208,16 @@ func FormatNetworkPolicyYAML(yaml string, clusterName string, funcMap template.F
 	return tpl.String(), nil
 }
 
+// ptrTypeIs checks if src is of type target. If src is a pointer or interface, it looks at its actual type
+// sprig has a `typeIs` and `kindIs` but it doesn't work cleanly with pointers
+func ptrTypeIs(target string, src interface{}) bool {
```

Comment:
Rather than creating a function that exposes a lower level reflect functionality, what do you think about introducing a function to check just what you need (e.g. `IsViolationKeyValue`). This at least avoids having to provide the type as string in the template, which can easily be missed in refactors and reference lookups.
 
```go
 func isViolationKeyValue(src interface{}) bool {
        r := reflect.TypeOf(src)
	for r.Kind() == reflect.Ptr || r.Kind() == reflect.Interface {
		r = r.Elem()
	}
        return r == reflect.TypeOf(storage.Alert_Violation_KeyValueAttrs_) // not sure if that's the right type
}
```

Reply:
Good call. And it can actually be made simpler too
```
func isViolationKeyValue(src interface{}) bool {
	return reflect.TypeOf(src) == reflect.TypeOf((*storage.Alert_Violation_KeyValueAttrs_)(nil))
}
```